labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -46,16 +50,23 @@ object PatternMatchingExercise {
    *        
    *    - anything else, the function result is "Some Scala class"
    */
-  def matchOnInputType(in: Any): String = {
-    error("fix me")
+  def matchOnInputType(in: Any): String = in match {
+    case s:String => "A string with length " + s.length()
+    case i:Int if i >0 => "A positive integer"
+    case p:Person => "A person with name: " + p.name
+    case s:Seq[Any] if s.length>10 => "Seq with more than 10 elements"
+    case s:Seq[Any] if s.length>2 => "first: " + s.head + ", second: " + s.tail.head + ", rest: " + s.tail.tail
+    case null => "A null value"
+    case _=> "Some Scala class"
```

Comment:
You can use Scala's string interpolation for this kind of thing. For example, instead of `"first: " + s.head`, you can write `s"first: ${s.head}"`. You can put whatever kind of expression you'd like in there, I believe.